File: src/render/markdown.ts
import type { Verdict } from "../types.js";

/** The verdict as INFRA.md, written into the repository that was analyzed. */
export function renderMarkdown(verdict: Verdict): string {
  const sections = [
    "# Infrastructure",
    "What this repository needs today, based only on the files in it.",
    `**Stage:** ${verdict.stage}`,
    `**Headroom:** ${verdict.headroom}`,
    `**Tripwire:** ${verdict.tripwire}`,
    verdict.flags.length > 0
      ? "**Flags:**\n\n" + verdict.flags.map((flag) => `- ${flag}`).join("\n")
      : "**Flags:** None.",
    verdict.confidenceNote,
  ];

  if (verdict.doNothingToday) {
    sections.push("Do nothing today.");
  }

  sections.push("Written by shiftpoint. Run `shiftpoint --write` to update.");

  return sections.join("\n\n") + "\n";
}

File: internal/render/markdown.go
package render

import (
	"strings"

	"github.com/MithrilBytes/shiftpoint/rules"
)

// Markdown renders a verdict as the INFRA.md written into the repository that
// was analyzed.
func Markdown(verdict rules.Verdict) string {
	flags := "**Flags:** None."
	if len(verdict.Flags) > 0 {
		var bullets []string
		for _, flag := range verdict.Flags {
			bullets = append(bullets, "- "+flag)
		}
		flags = "**Flags:**\n\n" + strings.Join(bullets, "\n")
	}

	sections := []string{
		"# Infrastructure",
		"What this repository needs today, based only on the files in it.",
		"**Stage:** " + verdict.Stage,
		"**Headroom:** " + verdict.Headroom,
		"**Tripwire:** " + verdict.Tripwire,
		flags,
		verdict.ConfidenceNote,
	}
	if verdict.DoNothingToday {
		sections = append(sections, "Do nothing today.")
	}
	sections = append(sections, "Written by shiftpoint. Run `shiftpoint --write` to update.")

	return strings.Join(sections, "\n\n") + "\n"
}
